ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -53,25 +53,58 @@ let sync_required ~__context ~host =
     end
   | _ -> raise Api_errors.(Server_error (internal_error, ["Cannot have more than one Cluster object per pool currently"]))
 
+let call_api_function_with_alert ~__context ~msg ~cls ~obj_uuid ~body ~(api_func : (Rpc.call -> Rpc.response) -> API.ref_session -> 'a) =
+  Helpers.call_api_functions ~__context
+    (fun rpc session_id ->
+       try
+         api_func rpc session_id |> ignore
+       with err ->
+         error "%s" ExnHelper.(string_of_exn err);
+         Xapi_alert.add ~msg ~cls ~obj_uuid ~body;
+         raise err
+    )
+
 let create_as_necessary ~__context ~host =
   match sync_required ~__context ~host with
   | Some cluster_ref -> (* assume pool autojoin set *)
     let network = get_network_internal ~__context ~self:cluster_ref in
     let (pifref,pifrec) = Xapi_clustering.pif_of_host ~__context network host in
     fix_pif_prerequisites ~__context (pifref,pifrec);
-    Helpers.call_api_functions ~__context (fun rpc session_id ->
-        Client.Client.Cluster_host.create rpc session_id cluster_ref host pifref) |> ignore
+    let api_func =
+      (fun rpc session_id ->
+         Client.Client.Cluster_host.create rpc session_id cluster_ref host pifref
+      ) in
+    let body =
+      (Printf.sprintf
+         "Unable to create cluster host on %s."
+         (Db.Host.get_name_label ~__context ~self:host)
```

Comment:
Minor comment: does this need to be split over so many lines? the parenthesis are not necessary, how long would it be if you put the `let body`, `Printf`, and format string on same line?